src/lib/es2015.collection.d.ts

```diff
@@ -58,14 +58,14 @@ interface WeakMap<K extends object, V> {
     has(key: K): boolean;
     /**
      * Adds a new element with a specified key and value.
-     * @param key Must be an object.
+     * @param key Must be an object or symbol.
      */
     set(key: K, value: V): this;
 }
 
 interface WeakMapConstructor {
-    new <K extends object = object, V = any>(entries?: readonly (readonly [K, V])[] | null): WeakMap<K, V>;
-    readonly prototype: WeakMap<object, any>;
+    new <K extends WeakKey = WeakKey, V = any>(entries?: readonly [K, V][] | null): WeakMap<K, V>;

```

Comment:
Regression from #50449.

Reply:
Ah, thanks for catching this. I missed it in the review.

Reply:
https://github.com/microsoft/TypeScript/pull/54592 will fix this